docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,282 @@
+
+Handling sensitive information, such as credentials is common between the two methods, see [Injecting credentials](#injecting-credentials).
+
+## Manage the configuration with Helm
+
+There are three ways configuration parameters can be modified:
+
+1. Copying the whole `mimir.config` value and modifying the configuration as text
+1. Setting parameters via the `mimir.structuredConfig` value (recommended)
+1. Setting extra CLI flags for components individually
+
+See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
+
+> **Limitation:**: it is not possible to delete configuration parameter settings via `mimir.structuredConfig` that were set in `mimir.config`. Set the configuration parameter to its default or to some other value instead.
+
+### How the configuration is applied
+
+Grafana Mimir components are run with a configuration calculcated by the following process:
+
+1. The configuration YAML in `mimir.config` is evaulated for Helm templates. This step is intended to ensure that the configuration applies to the Kubernetes cluster where it will be installed. For example setting up cluster specific addresses.
+1. The values from `mimir.structuredConfig` are merged on top and the result is again evaulated for Helm templates. This steps is intended to apply user specific customizations. For example S3 storage details.
+1. The resulting YAML configuration is then sorted alphabetically and stored in a `ConfigMap` (or `Secret` depending on the value of `configStorageType`) and provided to all Grafana Mimir components.
+1. When Grafana Mimir components are run in pods, the configuration file as well as any extra CLI flags are provided to the component.
+1. Each component evaulates the configuration, substituting environment variables as required. Note that extra CLI flags take precedence over the configuration file.
+
+> **Note:**: CLI flags are component specific, thus they will not show up in the generated `ConfigMap` (or `Secret`), making it less obvious what configuration is running. Use only when absolutely necessary.
+
+### Inspect changes to the configuration before upgrade
+
+Follow these steps to inspect what change will be applied to the configuration.
+
+Preparation:
+
+1. Install the [helm diff](https://github.com/databus23/helm-diff) plugin.
+1. Make sure to use `configStorageType` is set to `ConfigMap`
+
+Inspecting changes with the `helm diff` sub command:
+
+```bash
+helm -n mimir-test diff upgrade grafana/mimir-distributed -f custom.yaml
+```
+
+This command shows the differences between the running deployment and the deployment that would result from executing the `helm upgrade` command. Search for `name: mimir-config` in the output to see the difference in configuration settings.
+
+> **Note:** CLI flags and their difference are found in the `Deployment` and `StatefulSet` objects.
+
+## Manage the configuration directly
```

Comment:
"directly" is slightly confusing. If I am writing YAML in the `mimir.config` it's still directly managing what gets passed to mimir. I get what you mean by directly, but I prefer "externally" instead of "directly". This is also the language used in the chart (`externalSecretName`, `externalConfigVersion`, etc) and the headings in that section